java_tests/src/test/java/org/lsst/sal/SAL_Test_EvtLateJoinerNewestTest.java

```diff
@@ -0,0 +1,40 @@
+package org.lsst.sal;
+
+import Test.logevent_arrays;
+import org.junit.Assert;
+import org.junit.Test;
+
+public class SAL_Test_EvtLateJoinerNewestTest extends BaseTestCase {
+
+    /**
+     * Test that a late joiner can see an event using getEvent.
+     * <p>
+     * Only one value should be retrievable but it turns out that all are. That is why
+     * this test case is in a separate file because otherwise all data from all previously
+     * run unit tests are retrievable. It is unclear why this is the case.
```

Comment:
Do you set a random LSST_DDS_PARTITION_PREFIX before running each test? I don't see that happening here. That should isolate data from each test to only that test.

Reply:
In this test case I didn't implement it because it will always run in a separate JVM now. This means that it starts with a clean memory and no knowledge of any other unit tests. So, as the comment explains, that is why the test was implemented in a separate class (I corrected the JavaDoc).